new_audit: td-has-header

jazyan opened the pull request:
See http://go/prcpg for the current list of audits to be enabled.

Surprisingly, `a11y_tester.html` already has [a specific test case](https://github.com/GoogleChrome/lighthouse/blob/main/cli/test/fixtures/a11y/a11y_tester.html#L209-L217) for this.

connorjclark commented:
This audit has some funny history. #6169

connorjclark commented:
Can you fill out the expected audit result for `td-has-header` in `cli/test/smokehouse/test-definitions/a11y.js`? Ran via `yarn smoke a11y`

paulirish commented:
fwiw This is the test failure I'm seeing in the CI logs: 
<img width="1079" alt="image" src="https://user-images.githubusercontent.com/39191/236860952-c427ce09-11d3-4889-9766-cc92f7774532.png">

jazyan commented:
Added `td-has-header` case to a11y smoke tests, and updated the category renderer unit test. 

Looking into the DevTools e2e failing assertions 👀 

(and whoops, accidentally re-added commits after fetch from main.)

brendankenny commented on core/audits/accessibility/td-has-header.js:
```diff
@@ -0,0 +1,45 @@
+
+const UIStrings = {
+  /** Title of an accesibility audit that evaluates if all large table elements use the headers HTML attribute. This title is descriptive of the successful state and is shown to users when no user action is required. */
+  title: '`<td>` elements in a large `<table>` has one or more table headers.',
```
```suggestion
  title: '`<td>` elements in a large `<table>` have one or more table headers.',
```

brendankenny commented on core/audits/accessibility/td-has-header.js:
```diff
@@ -0,0 +1,45 @@
+  /** Title of an accesibility audit that evaluates if all large table elements use the headers HTML attribute. This title is descriptive of the successful state and is shown to users when no user action is required. */
+  title: '`<td>` elements in a large `<table>` has one or more table headers.',
+  /** Title of an accesibility audit that evaluates if all large table elements use the headers HTML attribute. This title is descriptive of the failing state and is shown to users when there is a failure that needs to be addressed. */
+  failureTitle: '`<td>` elements in a large `<table>` do not have one or more table headers.',
```
maybe 
```suggestion
  failureTitle: '`<td>` elements in a large `<table>` do not have table headers.',
```

brendankenny commented on core/audits/accessibility/td-has-header.js:
```diff
@@ -0,0 +1,45 @@
+  failureTitle: '`<td>` elements in a large `<table>` do not have one or more table headers.',
+  /** Description of a Lighthouse audit that tells the user *why* they should try to pass. This is displayed after a user expands the section to see more. No character length limits. The last sentence starting with 'Learn' becomes link text to additional documentation. */
+  description: 'Screen readers have features to make navigating tables easier. Ensuring ' +
+      'that `<td>` elements in a large (3 or more cells wide and high) table have an associated ' +
```
maybe
```suggestion
      'that `<td>` elements in a large (3 or more cells in width and height) table have an associated ' +
```
? Not sure if it's better but the first time I read the current version I had to re-read to get what the parenthetical was saying

brendankenny replied:
and maybe even
```suggestion
      'that `<td>` elements in a large table (3 or more cells in width and height) have an associated ' +
```
to keep the adjective with the noun, but really getting into the nits now

brendankenny reviewed (approved):
just some potential `UIStrings` suggestions but otherwise LGTM!
